### PR TITLE
Some attempted fixes for failing tests in the current master branch

### PR DESCRIPTION
**Pull request by apdavison**
(presumably ones which are skipped by Travis). Please could @maxgillett and @samuelgarcia take a look, as the failures were in `AxonIO` and `PlexonIO`.

### CODE REVIEW

File: neo/io/axonio.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def read_block(self, lazy=False, cascade=True):
 
                     anaSig = AnalogSignal(signal, sampling_rate=sampling_rate,
                                           t_start=t_start,
-                                          name=name.decode('utf-8'),
+                                          name=str(name),
```
Comment (samuelgarcia) [inline neo/io/axonio.py]: What happen if na me contain non ascii ?
better name.decode('utf-8').encode('ascii', 'ignore') no ?

File: neo/io/plexonio.py [hunk 2]
```diff
@@ -241,17 +241,18 @@ def read_segment(self, lazy=False, cascade=True, load_spike_waveform=True):
         for chan, h in iteritems(eventHeaders):
             if lazy:
                 times = []
+                labels = None
```
Comment (samuelgarcia) [inline neo/io/plexonio.py]: why not labels = [ ]

File: neo/io/axonio.py (hunk 1 above)
Comment (apdavison) [inline neo/io/axonio.py]: I'm not sure what the restrictions are on the type of `AnalogSignal.name`. The tests require it to be of type `str`, which means effectively ASCII for Python 2 and unicode for Python 3. Couldn't we allow unicode in Python 2?

Comment (samuelgarcia) [inline neo/io/axonio.py]: Yes. it would be better. It is not a big API break.
The thing is that we need to hack the test engine which test that each attribute have the good type (only one).
If we change AnalogSignal.name to unicode we need to change for all IOs. The better would to allow several types for each attributes whihch is not the case see AnalogSignal._necessary_attrs and BaseNeo._necessary_attrs.

File: neo/io/plexonio.py (hunk 2 above)
Comment (apdavison) [inline neo/io/plexonio.py]: The constructor for `EventArray` is :

```python
EventArray(times=None, labels=None, name=None, description=None,
                   file_origin=None, **annotations)
```

Comment (samuelgarcia) [inline neo/io/plexonio.py]: Yes but in case of lazy load the len need to be 0 because in case of full load labels is not None but an array.
This is specifyed nowhere but in for me in case of lazyload : each array need to be of size 0 when exist at full load.

The IO support labels for EventArray so labels = [ ].
For other IOs that have EventArray.labels = None even at fullload so when lazyload EventArray.labels = None.

I am not really sur to clear, sorry.

Comment (apdavison) [inline neo/io/plexonio.py]: But inside the `EventArray` constructor is the following:

```python
if labels is None:
    labels = np.array([], dtype='S')
```

so the result of passing `labels=None` is that `labels` becomes an empty array of strings.

Comment (samuelgarcia) [inline neo/io/plexonio.py]: OK sorry. i have not touch the code for a while.

File: neo/io/axonio.py (hunk 1 above)
Comment (apdavison) [inline neo/io/axonio.py]: ok, changing to unicode for all IOs is a big effort, so I think that should be done in a separate PR. If you're otherwise ok with these fixes, can you merge this?
